test/parallel/test-buffer.js

```diff
@@ -749,6 +749,15 @@ for (let i = 0; i < 256; i++) {
   assert.equal(hexb2[i], hexb[i]);
 }
 
+// Test single hex character throws TypeError
+// - https://github.com/nodejs/node/issues/6770
+assert.throws(function() {
+  Buffer.from('A', 'hex');
+}, TypeError);
+
+// Test single base64 char encodes as 0
+assert.equal(Buffer.from('A', 'base64'), 0);
```

Comment:
This should use `assert.strictEqual` and should test the length of the buffer rather than the buffer itself.